### PR TITLE
Allow config of additonal CircuitBreaker params

### PR DESCRIPTION
**Pull request by aidy**
### What does this PR do?
Allow the configuration of the other circuit breaker params mentioned in the documentation:
 * CheckPeriod
 * FallbackDuration
 * Recover(ing|y)Duration

Rename RecoveringDuration to RecoveryDuration to match the underlying library.

### Motivation

Fixes https://github.com/traefik/traefik/issues/3858

### More

- [ ] Added/updated tests
- [x] Added/updated documentation

### CODE REVIEW

**Review by rtribotte — APPROVED** [review]
Thanks 👍

File: docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml [hunk 1]
```diff
@@ -91,8 +91,25 @@ spec:
...
                     type: string
+                  fallbackDuration:
+                    description: FallbackDuration is the duration for which the circuit
+                      breaker will wait before trying to recover (tripped state).
+                    format: int64
+                    type: integer
+                  recoveryDuration:
+                    description: RecoveryDuration is the duration for which the circuit
+                      breaker will try to recover (recovering state).
+                    format: int64
```
Comment (aidy) [inline docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml:111]: I think these should be "anyOfs" like https://github.com/traefik/traefik/blob/fe684de5983d10efc93a1bb4f4e9b50f736e41e8/docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml#L553-L557

as they can take duration strings as well as a nanosecond int.

**Review by kevinpollet — APPROVED** [review]
Thanks 👍

File: docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml (hunk 1 above)
Comment (aidy) [reply docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml:111]: (I'm obviously happy to add a commit accordingly, but I don't know if that's going to break your process)

Comment (kevinpollet) [reply docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml:111]: Right, we will that 👍

File: pkg/middlewares/circuitbreaker/circuit_breaker.go [hunk 2]
```diff
@@ -31,19 +29,19 @@ func New(ctx context.Context, next http.Handler, confCircuitBreaker dynamic.Circ
 	logger.Debugf("Setting up with expression: %s", expression)
 
 	cbOpts := []cbreaker.CircuitBreakerOption{
-		createCircuitBreakerOptionExpression(expression),
+		createCircuitBreakerOptions(expression),
```
Comment (aidy) [inline pkg/middlewares/circuitbreaker/circuit_breaker.go:32]: hm, I'd deliberately renamed this as it only created the `Expression` option.

I did consider leaving it as `createCircuitBreakerOptions`, but then I think we should pass it the whole `confCircuitBreaker` struct, and have it produce all of the relevant options.

File: docs/content/middlewares/http/circuitbreaker.md [hunk 3]
```diff
@@ -171,15 +171,18 @@ This behavior cannot be configured.
 
 ### `CheckPeriod`
 
-The interval used to evaluate `expression` and decide if the state of the circuit breaker must change.
-By default, `CheckPeriod` is 100ms. This value cannot be configured.
+_Optional, Default="100ms"_
+
+The interval between successive checks of the circuit breaker condition (standby state).
```
Comment (mpl) [inline docs/content/middlewares/http/circuitbreaker.md:176]:
```suggestion
The interval between successive checks of the circuit breaker condition (when in standby state).
```

File: docs/content/middlewares/http/circuitbreaker.md [hunk 4]
```diff
@@ -171,15 +171,18 @@ This behavior cannot be configured.
...
 
 ### `FallbackDuration`
 
-By default, `FallbackDuration` is 10 seconds. This value cannot be configured.
+_Optional, Default="10s"_
+
+The duration for which the circuit breaker will wait before trying to recover (tripped state).
```
Comment (mpl) [inline docs/content/middlewares/http/circuitbreaker.md:182]:
```suggestion
The duration for which the circuit breaker will wait before trying to recover (from a tripped state).
```

File: docs/content/middlewares/http/circuitbreaker.md [hunk 5]
```diff
@@ -171,15 +171,18 @@ This behavior cannot be configured.
...
 
-### `RecoveringDuration`
+### `RecoveryDuration`
 
-The duration of the recovering mode (recovering state).
+_Optional, Default="10s"_
 
-By default, `RecoveringDuration` is 10 seconds. This value cannot be configured.
+The duration for which the circuit breaker will try to recover (recovering state).
```
Comment (mpl) [inline docs/content/middlewares/http/circuitbreaker.md:188]:
```suggestion
The duration for which the circuit breaker will try to recover (as soon as it is in recovering state).
```

File: docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml [hunk 6]
```diff
@@ -91,8 +91,31 @@ spec:
               circuitBreaker:
                 description: CircuitBreaker holds the circuit breaker configuration.
                 properties:
+                  checkPeriod:
+                    anyOf:
+                    - type: integer
+                    - type: string
+                    description: CheckPeriod is the interval between successive checks
+                      of the circuit breaker condition (standby state).
```
Comment (mpl) [inline docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml:99]:
```suggestion
                      of the circuit breaker condition (when in standby state).
```

File: docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml [hunk 7]
```diff
@@ -91,8 +91,31 @@ spec:
...
                   expression:
+                    description: Expression is the condition that triggers the tripped
+                      state.
                     type: string
+                  fallbackDuration:
+                    anyOf:
+                    - type: integer
+                    - type: string
+                    description: FallbackDuration is the duration for which the circuit
+                      breaker will wait before trying to recover (tripped state).
```
Comment (mpl) [inline docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml:110]:
```suggestion
                      breaker will wait before trying to recover (from a tripped state).
```

File: docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml [hunk 8]
```diff
@@ -91,8 +91,31 @@ spec:
...
+                    x-kubernetes-int-or-string: true
+                  recoveryDuration:
+                    anyOf:
+                    - type: integer
+                    - type: string
+                    description: RecoveryDuration is the duration for which the circuit
+                      breaker will try to recover (recovering state).
```
Comment (mpl) [inline docs/content/reference/dynamic-configuration/traefik.containo.us_middlewares.yaml:117]:
```suggestion
                      breaker will try to recover (as soon as it is in recovering state).
```

File: integration/fixtures/k8s/01-traefik-crd.yml [hunk 9]
```diff
@@ -535,8 +535,31 @@ spec:
               circuitBreaker:
                 description: CircuitBreaker holds the circuit breaker configuration.
                 properties:
+                  checkPeriod:
+                    anyOf:
+                    - type: integer
+                    - type: string
+                    description: CheckPeriod is the interval between successive checks
+                      of the circuit breaker condition (standby state).
```
Comment (mpl) [inline integration/fixtures/k8s/01-traefik-crd.yml:543]:
```suggestion
                      of the circuit breaker condition (when in standby state).
```

File: integration/fixtures/k8s/01-traefik-crd.yml [hunk 10]
```diff
@@ -535,8 +535,31 @@ spec:
...
                   expression:
+                    description: Expression is the condition that triggers the tripped
+                      state.
                     type: string
+                  fallbackDuration:
+                    anyOf:
+                    - type: integer
+                    - type: string
+                    description: FallbackDuration is the duration for which the circuit
+                      breaker will wait before trying to recover (tripped state).
```
Comment (mpl) [inline integration/fixtures/k8s/01-traefik-crd.yml:554]:
```suggestion
                      breaker will wait before trying to recover (from a tripped state).
```

File: integration/fixtures/k8s/01-traefik-crd.yml [hunk 11]
```diff
@@ -535,8 +535,31 @@ spec:
...
+                    x-kubernetes-int-or-string: true
+                  recoveryDuration:
+                    anyOf:
+                    - type: integer
+                    - type: string
+                    description: RecoveryDuration is the duration for which the circuit
+                      breaker will try to recover (recovering state).
```
Comment (mpl) [inline integration/fixtures/k8s/01-traefik-crd.yml:561]:
```suggestion
                      breaker will try to recover (as soon as it is in recovering state).
```

File: pkg/config/dynamic/middlewares.go [hunk 12]
```diff
@@ -93,7 +93,21 @@ type Chain struct {
 
 // CircuitBreaker holds the circuit breaker configuration.
 type CircuitBreaker struct {
+	// Expression is the condition that triggers the tripped state.
 	Expression string `json:"expression,omitempty" toml:"expression,omitempty" yaml:"expression,omitempty" export:"true"`
+	// CheckPeriod is the interval between successive checks of the circuit breaker condition (standby state).
```
Comment (mpl) [inline pkg/config/dynamic/middlewares.go:98]:
```suggestion
	// CheckPeriod is the interval between successive checks of the circuit breaker condition (when in standby state).
```

File: pkg/config/dynamic/middlewares.go [hunk 13]
```diff
@@ -93,7 +93,21 @@ type Chain struct {
...
+	CheckPeriod ptypes.Duration `json:"checkPeriod,omitempty" toml:"checkPeriod,omitempty" yaml:"checkPeriod,omitempty" export:"true"`
+	// FallbackDuration is the duration for which the circuit breaker will wait before trying to recover (tripped state).
```
Comment (mpl) [inline pkg/config/dynamic/middlewares.go:100]:
```suggestion
	// FallbackDuration is the duration for which the circuit breaker will wait before trying to recover (from a tripped state).
```

File: pkg/config/dynamic/middlewares.go [hunk 14]
```diff
@@ -93,7 +93,21 @@ type Chain struct {
...
+	FallbackDuration ptypes.Duration `json:"fallbackDuration,omitempty" toml:"fallbackDuration,omitempty" yaml:"fallbackDuration,omitempty" export:"true"`
+	// RecoveryDuration is the duration for which the circuit breaker will try to recover (recovering state).
```
Comment (mpl) [inline pkg/config/dynamic/middlewares.go:102]:
```suggestion
	// RecoveryDuration is the duration for which the circuit breaker will try to recover (as soon as it is in recovering state).
```

File: pkg/provider/kubernetes/crd/kubernetes.go [hunk 15]
```diff
@@ -425,6 +431,38 @@ func createPluginMiddleware(plugins map[string]apiextensionv1.JSON) (map[string]
...
+	if circuitBreaker == nil {
+		return nil, nil
+	}
+
+	cb := &dynamic.CircuitBreaker{Expression: circuitBreaker.Expression}
+	cb.SetDefaults()
+
+	if circuitBreaker.CheckPeriod != nil {
+		err := cb.CheckPeriod.Set(circuitBreaker.CheckPeriod.String())
+		if err != nil {
```
Comment (mpl) [inline pkg/provider/kubernetes/crd/kubernetes.go:444]:
```suggestion
		if err := cb.CheckPeriod.Set(circuitBreaker.CheckPeriod.String()); err != nil {
```

File: pkg/provider/kubernetes/crd/kubernetes.go [hunk 16]
```diff
@@ -425,6 +431,38 @@ func createPluginMiddleware(plugins map[string]apiextensionv1.JSON) (map[string]
...
+	if circuitBreaker.FallbackDuration != nil {
+		err := cb.FallbackDuration.Set(circuitBreaker.FallbackDuration.String())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if circuitBreaker.RecoveryDuration != nil {
+		err := cb.RecoveryDuration.Set(circuitBreaker.RecoveryDuration.String())
+		if err != nil {
```
Comment (mpl) [inline pkg/provider/kubernetes/crd/kubernetes.go:458]:
```suggestion
		if err := cb.RecoveryDuration.Set(circuitBreaker.RecoveryDuration.String()); err != nil {
```

**Review by mpl — APPROVED** [review]

**Pull request merged**